src/TestReport.ts

```diff
@@ -1,14 +1,133 @@
 import fs = require('fs');
 var mkdirp = require("mkdirp");
 
-interface message {
-    name?: string;
-    result?: boolean;
-    sent?: JSON;
-    received?: JSON;
-    errorId?: number;
-    error?: string;
+export class InteractionTestReportContainer {
+    testCycle: number;
+    testScenario: number;
+    name: string;
+    passed: boolean;
+
+    constructor(testCycle: number, testScenario: number, name: string) {
+        this.testCycle = testCycle;
+        this.testScenario = testScenario;
+        this.name = name;
+        this.passed = true;
+    }
+}
+
+export class Result {
+    id: number;
+    message?: string;
+
+    constructor(id: number, message?: string) {
+        this.id = id;
+        this.message = message;
+    }
+}
+
+export class Payload {
+    timestamp: Date;
+    payload: JSON;
+
+    constructor(timestamp: Date, payload: JSON) {
+        this.timestamp = timestamp;
+        this.payload = payload;
+    }
+}
+
+export class MiniTestReport {
+    passed: boolean | null; //If used for PropertyTesting each subTest (read, write, subscribe) can fail individually.
+    sent: Payload | null; //timestamp, sentMessage
+    received: Payload | null; //timestamp, receivedMessage
+    result: Result; //resultID, errorMessage
+
+    constructor(passed: boolean = null) {
+        this.passed = passed;
+        this.sent = null;
+        this.received = null;
+        this.result = null;
+    }
+}
+
+export class ActionTestReportContainer extends InteractionTestReportContainer {
+    report: MiniTestReport;
+
+    constructor(testCycle: number, testScenario: number, name: string) {
+        super(testCycle, testScenario, name);
+        this.report = new MiniTestReport();
+    }
+
+    /**
+     * Restructures the ReportContainer to match the structure of the printed Message. Potentially a new JSON object
+     * could be created and returned here if in the future the originally formatted container is needed for further processing.
+     * @return The restructured testReportContainer.
+     */
+    getPrintableMessage() {
```

Comment:
Why actually deleting them but not printing what is needed?

Reply:
The original container is not needed any further and this way a new object does not have to be created &rarr; less code is needed which reduces maintenance and improves readability. But i can create a new JSON object here if you want.